crates/ef-testing/src/evm_sequencer/evm_state.rs

```diff
@@ -191,6 +197,50 @@ impl EvmState for KakarotSequencer {
             Into::<FieldElement>::into(nonce).to_bytes_be(),
         ))
     }
+
+    fn get_code_at(&mut self, evm_address: &Address) -> StateResult<Bytes> {
+        let starknet_address = compute_starknet_address(evm_address);
+
+        let bytecode_len = (&mut self.0.state).get_storage_at(
+            starknet_address.try_into()?,
+            get_storage_var_address("bytecode_len_", &[])?,
+        )?;
+        let bytecode_len: u64 = bytecode_len.try_into()?;
+        if bytecode_len == 0 {
+            return Ok(Bytes::default());
+        }
+
+        let full_16 = bytecode_len / 16;
+        let mut bytecode: Vec<u8> = Vec::new();
+
+        for i in 0..full_16 {
+            let key = get_storage_var_address("bytecode_", &[StarkFelt::from(i)])?;
+            let code = (&mut self.0.state).get_storage_at(starknet_address.try_into()?, key)?;
+            bytecode.append(&mut felt_to_bytes(&code.into(), 16).to_vec());
+        }
+
+        let remainder = bytecode_len % 16;
+        let key = get_storage_var_address("bytecode_", &[StarkFelt::from(full_16)])?;
+        let code = (&mut self.0.state).get_storage_at(starknet_address.try_into()?, key)?;
+        bytecode.append(&mut felt_to_bytes(&code.into(), remainder as usize).to_vec());
+
+        Ok(Bytes::from(bytecode))
+    }
+
+    fn get_balance_at(&mut self, evm_address: &Address) -> StateResult<U256> {
```

Comment:
maybe add a rustdoc to say this gets the balance of the native token (starknet_eth) of the underlying chain